### PR TITLE
Address #256

### LINKED ISSUE #256: can not show mountpoint for nvme disk
when I use ghw in my project on a machine with nvme, it just can not show nvme disk mountpoint.
I just run the sample code as
```
package main

import (
	"fmt"

	"github.com/jaypipes/ghw"
)

func main() {
	block, err := ghw.Block()
	if err != nil {
		fmt.Printf("Error getting block storage info: %v", err)
	}

	fmt.Printf("%v\n", block)

	for _, disk := range block.Disks {
		fmt.Printf(" %v\n", disk)
		for _, part := range disk.Partitions {
			fmt.Printf("  %v\n", part)
		}
	}
}
```
Output
```
block storage (3 disks, 12TB physical storage)
 nvme0n1 SSD (4TB) NVMe [@pci-0000:83:00.0-nvme-1 (node #0)] vendor=unknown model=P5920CH0384M00 serial=SH205002054 WWN=eui.000000000000000138b19e72c8000806
 sda HDD (447GB) SCSI [@pci-0000:05:00.0-scsi-0:2:0:0 (node #0)] vendor=AVAGO model=MR9361-8i serial=600605b00fa0be4027f9a6220325d7f3 WWN=0x600605b00fa0be4027f9a6220325d7f3
  sda1 (447GB) [ext4] mounted@/
 sdb HDD (8TB) SCSI [@pci-0000:05:00.0-scsi-0:2:1:0 (node #0)] vendor=AVAGO model=MR9361-8i serial=600605b00fa0be4027f9a6310400af84 WWN=0x600605b00fa0be4027f9a6310400af84
  sdb1 (8TB) [ext4] mounted@/mnt/data
```
lsblk output
```
NAME    MAJ:MIN RM   SIZE RO TYPE MOUNTPOINT
sda       8:0    0 446.6G  0 disk
└─sda1    8:1    0 446.6G  0 part /
sdb       8:16   0   7.3T  0 disk
└─sdb1    8:17   0   7.3T  0 part /mnt/data
nvme0n1 259:0    0   3.5T  0 disk /data
```
os information
```
NAME="Ubuntu"
VERSION="18.04.5 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.5 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
```
Did I use the code correctly ?

### PR DESCRIPTION
**Pull request by taigrr**
If a disk has no partitions, common use cases for looping over partitions on the disk array created by Block.New() don't work as expected.
Now, underlying partition functions detect partitionless disks and treat a full-disk filesystem as a partition.

The Partition UUID is set to the empty string (since there isn't one), so this behavior can be detected.

This issue will resolve #256

### CODE REVIEW

**Comment by taigrr** [comment]
@fromanirh Could you take a look at this please?

File: pkg/block/block_linux.go [hunk 1]
```diff
@@ -206,6 +206,23 @@ func diskPartitions(ctx *context.Context, paths *linuxpath.Paths, disk string) [
 		}
 		out = append(out, p)
 	}
+	// this is a disk with no partitions on it.
+	// Consider the filesystem on the disk to be a partition.
```
Comment (fromanirh) [inline pkg/block/block_linux.go:212]: codewise looks good. I'd like to have some (at least one) tests to cover this code before we can merge.

File: pkg/block/block_linux.go [hunk 2]
```diff
@@ -179,7 +179,9 @@ func diskWWN(paths *linuxpath.Paths, disk string) string {
 // diskPartitions takes the name of a disk (note: *not* the path of the disk,
 // but just the name. In other words, "sda", not "/dev/sda" and "nvme0n1" not
 // "/dev/nvme0n1") and returns a slice of pointers to Partition structs
-// representing the partitions in that disk
+// representing the partitions in that disk. If a disk has no partitions,
+// it returns information on the disk instead, but the Partition UUID is set
+// to the empty string (since there isn't one.)
```
Comment (fromanirh) [inline pkg/block/block_linux.go:184]: thanks for fixing!

**Review by fromanirh — COMMENTED** [review]
The code per se looks fine and it is very much like how I'd solve this bug. I think we should have some (at least one) test to exercise the new code path though, also to prevent future regressions.

File: pkg/block/block_linux.go (hunk 1 above)
Comment (taigrr) [reply pkg/block/block_linux.go:212]: Absolutely. However, I will admit after looking through the block tests, I'm not sure I understand the project's approach to tests. I don't see any mocking, so in order to run a full test suite someone would have to use physical hardware to replicate this? What are the chances someone will use a partition-less disk in testing?

Or are there mocks I'm missing and I'm just being a dunce 😆

Comment (taigrr) [reply pkg/block/block_linux.go:212]: Hey @fromanirh can you give me a pointer to an example test to mimic? I'd like to wrap this up.

Comment (taigrr) [reply pkg/block/block_linux.go:212]: @fromanirh hate to bother, but can I bump this? Would like to know what to add to get this merged.

Comment (fromanirh) [reply pkg/block/block_linux.go:212]: @taigrr sure thing, sorry for the dealy. On it.

Comment (fromanirh) [reply pkg/block/block_linux.go:212]: > Absolutely. However, I will admit after looking through the block tests, I'm not sure I understand the project's approach to tests. I don't see any mocking, so in order to run a full test suite someone would have to use physical hardware to replicate this? What are the chances someone will use a partition-less disk in testing?
> 
> Or are there mocks I'm missing and I'm just being a dunce laughing

Refactoring a bit the code to enable fake disks would work for me (cc @jaypipes though). The other option would be to actually capture the data reported by the system from a partitionless disk, using for example ghw-snapshot. A USB stick should be easy to set up and perfectly fit for the case.

Comment (fromanirh) [reply pkg/block/block_linux.go:212]: another option would be to have mock data under `testdata/'. This option sits between refactoring the code to enable unit tests and capturing real data with `ghw-snapshot`. I'd be fine with this approach as well

Comment (taigrr) [reply pkg/block/block_linux.go:212]: Great, now I perfectly understand the testing setup. I'll get back on this soon.

**Comment by taigrr** [comment]
I am closing this PR since github doesn't let you change the name of the source branch for comparison an I now have multiple PRs. I'll reopen and reference this PR in the new one.

File: pkg/block/block_linux.go (hunk 1 above)
Comment (taigrr) [reply pkg/block/block_linux.go:228]: @fromanirh I've closed this PR in favor of a dedicated branch, Please see #308 which adds a unit test for this.

**Pull request closed without merging**